rfcs/text/semantic-release.md

```diff
@@ -0,0 +1,257 @@
+- typically we will release from `master`
+- every commit to `master` may trigger a release depending upon the commit message
+- a merge commit will trigger **ONE** release, even if it has multiple features within
+- a machine user `carbonci` will publish the release to GitHub releases and [NPM](https://www.npmjs.com/)
+- `carbonci` will comment on all PR's stating the version that the changes are included in
+- `carbonci` will comment on all referenced issues stating the version that the issue was fixed in
+
+We will use the [`conventional-commits`](https://www.conventionalcommits.org/en/v1.0.0/) commit message format.
+- we will use the `conventionalcommits` preset for [`@semantic-release/commit-analyzer`](https://github.com/semantic-release/commit-analyzer)
+> The Conventional Commits specification is a lightweight convention on top of commit messages. It provides an easy set of rules for creating an explicit commit history; which makes it easier to write automated tools on top of. This convention dovetails with [SemVer](https://semver.org/), by describing the features, fixes, and breaking changes made in commit messages.
+>
+> The commit message should be structured as follows:
+> ```
+><type>[optional scope]: <description>
+>
+>[optional body]
+>
+>[optional footer(s)]
+>```
+> The commit contains the following structural elements, to communicate intent to the consumers of your library:
+>
+> 1. **fix:** a commit of the *type* `fix` patches a bug in your codebase (this correlates with [`PATCH`](https://semver.org/) in semantic versioning).
+> 1. **feat:** a commit of the *type* `feat` introduces a new feature to the codebase (this correlates with [`MINOR`](https://semver.org/) in semantic versioning).
+> 1. **BREAKING CHANGE:** a commit that has a footer `BREAKING CHANGE:`, or appends a `!` after the type/scope, introduces a breaking API change (correlating with [`MAJOR`](https://semver.org/) in semantic versioning). A BREAKING CHANGE can be part of commits of any *type*.
+> 1. *types* other than `fix:` and `feat:` are allowed, for example [`@commitlint/config-conventional`](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional) (based on the the [Angular convention](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#-commit-message-guidelines)) recommends `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others.
+> 1. footers other than `BREAKING CHANGE: <description> `may be provided and follow a convention similar to [git trailer format](https://git-scm.com/docs/git-interpret-trailers).
+>
+> Additional types are not mandated by the conventional commits specification, and have no implicit effect in semantic versioning (unless they include a BREAKING CHANGE).
+>
+> A scope may be provided to a commit’s type, to provide additional contextual information and is contained within parenthesis, e.g., `feat(parser): add ability to parse arrays`.  
```

Comment:
```suggestion
> A scope may be provided to a commit’s type (to provide additional contextual information) and must be contained within parentheses. For example: `feat(parser): add ability to parse arrays`.
```

Reply:
This is quoted, I would like to keep the original formatting.